synapse/lib/splice.py

```
@@ -0,0 +1,96 @@
+import tempfile
+
+import synapse.common as s_common
+import synapse.lib.msgpack as s_msgpack
+
+_readsz = 10000000
+
+
+def splice(act, **info):
+    '''
+    Form a splice event from a given act name and info.
+
+    Args:
+        act (str): The name of the action.
+        **info:    Additional information about the event.
+
+    Example:
+        splice = splice('add:node', form='inet:ipv4', valu=0)
+        self.fire(splice)
+
+    Notes:
+        Splice events were reworked in v0.0.45 and now contain a sub-event of
+        the (act, info) under the 'mesg' key.
+
+    Returns:
+        (str, dict): The splice event.
+    '''
+    return ('splice', {'mesg': (act, info)})
+
+def convertOldSplice(mesg):
+    '''
+    Converts an "old" splice event to the "new" format.
+
+    Args:
+        mesg ((str,dict)):  An event tuple.
+
+    Example:
+        newsplice = convertOldSplice(oldsplice)
+
+    Returns:
+        (str, dict): The splice event.
+    '''
+    if not(isinstance(mesg, tuple) and len(mesg) is 2):
+        return  # not a valid event
+
+    evtname = mesg[0]
+    if evtname != 'splice':
+        return  # not a splice
```

raise rather than return here

---

fixed